src/ble/BtpEngine.h

```diff
@@ -136,13 +137,14 @@ class BtpEngine
 
     BLE_ERROR HandleCharacteristicReceived(System::PacketBufferHandle data, SequenceNumber_t & receivedAck, bool & didReceiveAck);
     bool HandleCharacteristicSend(System::PacketBufferHandle data, bool send_ack);
-    BLE_ERROR EncodeStandAloneAck(PacketBuffer * data);
+    BLE_ERROR EncodeStandAloneAck(const PacketBufferHandle & data);
 
-    PacketBuffer * RxPacket();
-    PacketBuffer * TxPacket();
-
-    bool ClearRxPacket();
-    bool ClearTxPacket();
+    void ClearRxPacket();
+    PacketBufferHandle TakeRxPacket() { return std::move(mRxBuf); }
```

Comment:
Or perhaps what should happen is that `TakeRxPacket` should do all the state-maintenance and `ClearRxPacket` should just `TakeRxPacket` and ignore the return, or stick it into a stack var that then goes out of scope or something.

Reply:
It would be simpler. The reason I left it is that there is one instance in `BLEEndPoint::DriveSending()` where `mBtpEngineTest.DoTxTiming()` uses the taken packet, and that `CHIP_ENABLE_CHIPOBLE_TEST` code currently isn't in tree so I can't tell whether it also depends on `mTxState`, and I don't know whether someone intends to import that test code.

Reply:
Hmm.  I don't know that this `DoTxTiming` thing is even going to exist in CHIP.  That said, the Weave version doesn't depend on the engine's `mTxState` (note that it's being called on a different object: the test object, not the engine object).  So I think it's safe to make this change.  I'm OK with it being in this PR or in a followup, whichever works better for you.

Reply:
OK. I'll add it to the next PR.